src/components/form/input/Input.tsx

```diff
@@ -0,0 +1,88 @@
+/// <reference types="vite-plugin-svgr/client" />
+
+import Typography from "@/components/typography/Typography";
+import EyeClosed from "@/assets/eye-closed.svg?react";
+import EyeOpen from "@/assets/eye-open.svg?react";
+import React, { useState, forwardRef, useCallback } from "react";
+
+interface InputProps {
+  type: string;
+  label?: string;
+  placeholder?: string;
+  error?: string;
+  autoComplete?: string;
+  className?: string;
+}
+
+interface ToggleComponent {
+  (inputState: InputState, toggleShowIcon: (type: string) => void): JSX.Element;
+}
+
+interface InputState {
+  [key: string]: boolean; // Key is the toggle type (e.g., "password"), value is the show/hide state
+}
+
+const typeToToggleMap: { [key: string]: ToggleComponent } = {
```

Comment:
Nice! 
But isn't too complicate? maybe preferably do it a bit more `KIS` and just have another `ShowHideInput`.. Not really sure on this comment - to your judgment

Reply:
Anyway I would maybe transfer it into a different file